packages/uicore/src/components/ExpressionInput/ExpressionInput.tsx

```diff
@@ -371,12 +372,27 @@ export function ExpressionInput(props: ExpressionInputProps): React.ReactElement
           />
         )}
         <React.Fragment>
-          {listProps.itemList
-            ? React.cloneElement(listProps.itemList as React.ReactElement, {
-                className: css.menu,
-                style: { maxHeight }
-              })
-            : null}
+          {newExpressionComponent ? (
+            <NewExpressionDropdown
+              query={queryValue}
+              rootTrieNode={getDataStructureInCorrectFormat(items)}
```

Comment:
formatData or something with format would be better naming here instead of CorrectFormat

Reply:
done